File: silver/10828/c++/main.cpp
#include <iostream>
#include <stack>
#include <string>

int main()
{
    int num, data;
    std::stack<int> _stack;
    std::string command;
    std::cin >> num;

    for (int i = 0; i < num; i++) {
        std::cin >> command;
        if (command == "push") {
            std::cin >> data;
            _stack.push(data);
        } else if (command == "pop") {
            if (_stack.empty()) {
                std::cout << "-1\n";
            } else {
                std::cout << _stack.top() << '\n';
                _stack.pop();
            }
        } else if (command == "size") {
            std::cout << _stack.size() << '\n';
        } else if (command == "empty") {
            std::cout << _stack.empty() << '\n';
        } else if (command == "top") {
            if (_stack.empty()) {
                std::cout << "-1\n";
            } else {
                std::cout << _stack.top() << '\n';
            }
        }
    }

    return 0;
}

File: silver/10828/python/main.py
import sys
input = sys.stdin.readline

if (__name__ == "__main__"):
    num = int(input())
    stack = []
    for i in range(num):

        command = list(input().split())
        if (len(command) == 1):
            if (command[0] == "pop" and len(stack) > 0):
                print(stack.pop(-1))
            elif (command[0] == "size"):
                print(len(stack))
            elif (command[0] == "empty"):
                if (len(stack) == 0): print(1)
                else : print(0)
            elif (command[0] == "top" and len(stack) > 0):
                print(stack[-1])
            else :
                print(-1)
        else :
            stack.append(int(command[1]))